src/Components/SuggestionsSideBar/SuggestionsSideBar.jsx

```diff
@@ -7,40 +7,67 @@ import ListItemAvatar from "@mui/material/ListItemAvatar";
 import Avatar from "@mui/material/Avatar";
 import { Button } from "@mui/material";
 import { Box } from "@mui/material";
-import { useSelector } from "react-redux";
+import { useSelector, useDispatch } from "react-redux";
+import { followUser } from "../../store/userSlice";
+import { useNavigate } from "react-router-dom";
+import toast from "react-hot-toast";
 
 export const SuggestionsSideBar = () => {
   const { users } = useSelector((state) => state.users);
+  const { userInfo, token } = useSelector((state) => state.auth);
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  // filter user on the basis of user name for follow/ unfollow
+
+  const currentUser = users?.find(
+    (each) => each.username === userInfo.username
+  );
+
+  const currentUserFollowing = currentUser?.following;
+
+  const userSuggestion = [...users].filter(
```

Comment:
I guess you don't need to make a copy of  `users `. `users.filter()` will return a new array anyway.